### PR TITLE
Bug 550956 - [Passage] Support mail attachment

### PR DESCRIPTION
**Pull request by serjiokov**
The eml file creating feature provided from wizard license detailed
page.

Currently supported two mail option:
- mail body preparation
- .eml file preparation (with attachment)

### CODE REVIEW

**Comment by ruspl-afed** [comment]
You need to change commit message for ALL your commits and add 

Signed-off-by: your name <email.registered@eclipse>

File: bundles/org.eclipse.passage.lic.api/src/org/eclipse/passage/lic/api/mail/LicenseMailBuilder.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.lic.api.mail;
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.api/src/org/eclipse/passage/lic/api/mail/LicenseMailBuilder.java]: lic.api contains very basic concepts of passage, lets extract the email-related interfaces to lic.net

Also interface itself needs rework: "details" and "toText" should not depend on particular implementation, while the way to save as ".eml" should

I would expect something like 
```java
public void writeEml(MailDescriptor descriptor, OutputStream output) throws LicensingException
```

File: bundles/org.eclipse.passage.lic.mail/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -0,0 +1,15 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Version: 0.5.0.qualifier
+Bundle-Name: %Bundle-Name
+Bundle-Vendor: %Bundle-Vendor
+Bundle-Copyright: %Bundle-Copyright
+Bundle-SymbolicName: org.eclipse.passage.lic.mail
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.mail/META-INF/MANIFEST.MF]: Let's move mail interfaces inside lic.net and use this one to isolate javax.mail dependency

File: bundles/org.eclipse.passage.lic.mail/OSGI-INF/org.eclipse.passage.lic.mail.LicenseMailManagerImpl.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" name="org.eclipse.passage.lic.mail">
+	<implementation class="org.eclipse.passage.lic.internal.mail.LicenseMailManagerImpl"/>
+	<reference bind="bindLicenseMailBuilder" cardinality="0..1" interface="org.eclipse.passage.lic.api.mail.LicenseMailBuilder" name="LicenseMailBuilder" policy="static" unbind="unbindLicenseMailBuilder"/>
+	<service>
+		<provide interface="org.eclipse.passage.lic.mail.LicenseMailManager"/>
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.mail/OSGI-INF/org.eclipse.passage.lic.mail.LicenseMailManagerImpl.xml]: Having interface and its implementation in the same bundle makes impossible to switch implementation

File: bundles/org.eclipse.passage.lic.mail/src/org/eclipse/passage/lic/mail/LicenseMailManager.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ *     ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.lic.mail;
+
+import org.eclipse.passage.lic.api.mail.LicenseMailBuilder;
+
+/**
+ * The Licensing mail manager interface
+ * 
+ * @since 0.5.0
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.mail/src/org/eclipse/passage/lic/mail/LicenseMailManager.java]: Please setup API Baseline

File: bundles/org.eclipse.passage.lic.mail/src/org/eclipse/passage/lic/mail/LicenseMailManager.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ */
+public interface LicenseMailManager {
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.mail/src/org/eclipse/passage/lic/mail/LicenseMailManager.java]: Interface needs rework. This one sounds too fundamental: we are not managing emails, we are trying to generate the .eml file, right?

File: bundles/org.eclipse.passage.lic.mail/OSGI-INF/l10n/bundle.properties [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#Properties file for org.eclipse.passage.lic.api
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.mail/OSGI-INF/l10n/bundle.properties]: correct bundle symbolic name

Comment (serjiokov) [reply bundles/org.eclipse.passage.lic.mail/OSGI-INF/l10n/bundle.properties]: Done.

**Pull request closed without merging**